### DISCUSSION TITLE
Add missing features to Philiphs XT machines

### DISCUSSION THREAD

**Discussion by Fenix770** [86Box/86Box#4167] [Ideas]
**Is your feature request related to a problem? Please describe.**

No. When was first implemented on 86box, these machine could  change the clock speed only with the SPEED.COM utility included in Philipihs MS-DOS 3.20 and 3.30. 

A.The Philiphs P3120 starts in "turbo mode" (10 Mhz). Can be changet to 4.77 Mhz with "SPEED STANDARD" command.
B.The Philiphs P3105 starts in standard mode ,at 4.77 Mhz  at can changed to 7.16 in "turbo mode".

2. In a Youtube video about the Philiphs XT P3120 (it has BIOS version 3.16 and counts 768KB of RAM at startup.
On 86box memory counts stops at 640kb even when 768kb installed but I don't know if extra hardware emulation is required.

3. There is also a CLOCK.COM utility, but does not works with any RTC emulated in 86box. It seems to require some kind of internal clock.

[https://www.youtube.com/watch?v=3T5b4CJshAM](url) (memory count finish with 768kb at 2:26)

**Describe the solution you'd like**

1.The capability to change the CPU clock with SPEED utility inluded in Philiphs MS-DOS, and not only from 86box configuration.
2.Add the capability to have 768kb, if it's possible.

**Edit:**According the video is posible to install up to 1mb of RAM in this machine.

Hard disk file:  (It has dutch MS-DOS 3.2)
[Philiphs XTA Hard disk.zip](https://github.com/86Box/86Box/files/6805047/Philiphs.XTA.Hard.disk.zip)

Screenshoots:
![Philips XT](https://user-images.githubusercontent.com/31600563/125370619-37a1ad80-e355-11eb-8c4b-b203437524c0.jpg)
![Philips XT-2](https://user-images.githubusercontent.com/31600563/125370621-396b7100-e355-11eb-83e7-ae60ae38da81.jpg)

**Comment by EngiNerd89** [comment]
Let me clarify some aspects on your requests:

1) Speed selection has been implemented in both machines. However, please note that the mechanism to select the speed is different: Philips P3105 expects bit 7 at address 0x0c0 to be high in order to toggle turbo mode. This can be done dynamically with the SPEED utility. Also, by pressing CTRL-ALT-INS the machines reboots in turbo mode, whereas with CTRL-ALT-DEL the machine reboots in standard (4.7 MHz) mode. That said, I don't know if the machine is expected to cold-boot in standard or in turbo mode.
Conversely, Philips P3120 relies on the GC100A chipset to toggle turbo mode. In particular, when bit 0 at address 0x0c2 is set to high, turbo mode is activated. Thus, it should rely on a different SPEED utility than the one provided with P3105 disks (which I don't have). Also, the key combinations used to reboot the machine work the other way around: CTRL-ALT-INS reboots the machine in standard (4.7 MHz) mode, whereas CTRL-ALT-DEL reboots it in turbo mode.

2) Unfortunately I don't know how the 640-768 KB memory is handled by those machines, and if it is made available to the operating system as EMS or as HMA.

3) Both machines rely on a PCF8573T clock chip to store and maintain date and time information. Based on the behavior of the CLOCK utility, it appears that bits 4-5 at address 0x0c0 drive the I2C bus where the clock chip is connected. I found out that MAME has implemented the PCF8573T clock chip, so it could be ported to 86Box. However, I have yet to familiarize on how the I2C bus is implemented in 86Box (I see it has been used in some video cards).

**Comment by Fenix770** [comment]
Hello!

1.When the patch was first implemented (I don't remember the commit) the SPEED utility worked on both machines. What I'd like to be implemented is the possiblity to change the clock speed with the SPEED program under MS-DOS.

2. I don't think the extra memory can be used without the full documentation of these functions. The machine on the video is a P3105  and count 768kb of RAM at boot time think it may be used as conventional RAM. Unlike other XT machines  with EMS support like Toshiba T1000/1200. These counts only 640kb at boot and the rest is counted only by the EMS.SYS driver at boot time.I don't think HMA can be supported, it is the first 64kb of extended memory and that would require a 286.

3.It would be great to port it to 86box if it is possible.

Thanks!

**Comment by rderooy** [comment]
I actually had an NMS9110 back in the day.

From what I recall (my memory is a bit fuzzy on this), the extra 128kB could be used to load high device drivers, such as the mouse driver, using a special driver (the equivalent of a LoadHigh in later DOS versions) or could be used as a RAM disk. The utilities for this were on the provided disks.

**Comment by ghost** [comment]
The P3120 used non internal ega or vga card based on chips f82c451. 
Here is the links https://youtu.be/3T5b4CJshAM?t=671
Datasheet and rom only from octek jovian logic and gw-vga and the philips still undumped
https://www.vgamuseum.info/index.php/news/item/178-chips-technologies-f82c451

**Comment by rderooy** [comment]
The NMS9110 used an ATI 8bit ISA video card that could do Hercules and CGA, connected to a monochrome monitor. There was a utility to switch between the modes. All other ports were integrated into the systemboard. Except for the HDD controller (we had an ISA hardcard).

I guess it must have been the ATI Graphics Solution card (presumably the ATI CW16800-B), although I don't recall the plantronics mode.

https://classic.technology/wp-content/uploads/2021/05/philipsnms9100.pdf
Strangely, the PDF says, max memory was 640kB. But I have only ever seen 768kB units.

In this Spanish magazine article on page 13, it is listed as 640kB + 128kB RAMDISK
https://loadzx.com/wp-content/uploads/simple-file-list/LITERATURA/REVISTAS/RS232/RS232-n_o-3.pdf
